### PR TITLE
initialize ui-designer backend source code

### PR DESCRIPTION
**Pull request by cpuy**

### CODE REVIEW

**Comment by cpuy** [comment]
test this please

**Comment by cpuy** [comment]
retest this please

**Comment by javamind** [comment]
Why copyright files are now on the backend side. We won't have declaration in frontend source files ?

**Comment by cpuy** [comment]
No we don't have copyrights on frontend side for now.

**Comment by javamind** [comment]
We have some references to fragment 

In the Js resources use by generator (preview and export)
image

In i18n files
image

**Comment by cpuy** [comment]
Hum... yes.
- angular-mocks.js is not of our own code
- I'll remove po files and we'll get fresh ones from crowdin later
- i'll remove it from pbModel
- Let me see with @velcrin how we can do for generator.js

File: backend/webapp/bower.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{
+  "name": "page-builder-backend-webapp",
+  "version": "0.0.1",
```
Comment (javamind) [inline backend/webapp/bower.json]: Should be 1.0.0. In pom.xml we have 
```<version>1.0.0-SNAPSHOT</version>```

File: backend/webapp/package.json [hunk 2]
```diff
@@ -0,0 +1,45 @@
+{
+  "name": "page-builder-backend",
+  "version": "0.0.1",
```
Comment (javamind) [inline backend/webapp/package.json]: see above for the version number

File: backend/webapp/src/main/generator/js/generator.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+(function() {
+  'use strict';
+
+  angular.module('org.bonitasoft.pagebuilder.generator.constants', []);
+  angular.module('org.bonitasoft.pagebuilder.generator.services', []);
+  angular.module('org.bonitasoft.pagebuilder.generator.directives', []);
+  angular.module('org.bonitasoft.pagebuilder.fragments', []);
+  angular.module('org.bonitasoft.pagebuilder.widgets', []);
```
Comment (javamind) [inline backend/webapp/src/main/generator/js/generator.js]: pagebuilder ?

File: backend/webapp/src/main/java/org/bonitasoft/web/designer/controller/exception/ServerImportException.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+ * See the GNU Lesser General Public License for more details.
+ * You should have received a copy of the GNU Lesser General Public License along with this
+ * program; if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth
+ * Floor, Boston, MA 02110-1301, USA.
+**/
+package org.bonitasoft.web.designer.controller.exception;
+
+public class ServerImportException extends ImportException {
+
+    private static final long serialVersionUID = 3261567462198933160L;
```
Comment (javamind) [inline backend/webapp/src/main/java/org/bonitasoft/web/designer/controller/exception/ServerImportException.java]: not necessary

File: backend/webapp/src/main/java/org/bonitasoft/web/designer/experimental/parametrizedWidget/DatePickerWidget.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+ * program; if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth
+ * Floor, Boston, MA 02110-1301, USA.
+**/
+package org.bonitasoft.web.designer.experimental.parametrizedWidget;
+
+public class DatePickerWidget extends InputWidget {
+
+    private static final String DATE_PICKER_WIDGET_ID = "pbDatePicker";
+    private String dateFormat;
+    private boolean forceUTC = false;
```
Comment (javamind) [inline backend/webapp/src/main/java/org/bonitasoft/web/designer/experimental/parametrizedWidget/DatePickerWidget.java]: not used anymore

File: backend/webapp/src/main/java/org/bonitasoft/web/designer/repository/exception/NotAllowedException.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+ * This library is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ * You should have received a copy of the GNU Lesser General Public License along with this
+ * program; if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth
+ * Floor, Boston, MA 02110-1301, USA.
+**/
+package org.bonitasoft.web.designer.repository.exception;
+
+@SuppressWarnings("serial")
```
Comment (javamind) [inline backend/webapp/src/main/java/org/bonitasoft/web/designer/repository/exception/NotAllowedException.java]: Used by eclipse

File: backend/webapp/src/main/java/org/bonitasoft/web/designer/rendering/GenerationException.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ * You should have received a copy of the GNU Lesser General Public License along with this
+ * program; if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth
+ * Floor, Boston, MA 02110-1301, USA.
+**/
+package org.bonitasoft.web.designer.rendering;
+
+@SuppressWarnings("serial")
+public class GenerationException extends RuntimeException {
```
Comment (javamind) [inline backend/webapp/src/main/java/org/bonitasoft/web/designer/rendering/GenerationException.java]: Used by eclipse

File: backend/webapp/src/main/java/org/bonitasoft/web/designer/workspace/WorkspacePathResolver.java [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+ * Resolve bonita page designer workspace Path.
+ * Workspace directory should be specified via a system property <b>workspace</b>.
+ * If not, default value is {user.home}/.bonita
+ *
+ * Each repository folder can be changed as well by changing <b>repository.widgets</b>, <b>repository.pages</b>
+ *
+ * @author Colin Puy
+ */
+@Named("workspacePathResolver")
+public class WorkspacePathResolver {
```
Comment (javamind) [inline backend/webapp/src/main/java/org/bonitasoft/web/designer/workspace/WorkspacePathResolver.java]: "workspacePathResolver" is not necessary

**Comment by cpuy** [comment]
fragment module declaration done in https://github.com/bonitasoft/bonita-form-builder/pull/942
I'll remove it from community code

File: backend/webapp/src/main/java/org/bonitasoft/web/designer/repository/exception/NotAllowedException.java [hunk 9]
```diff
@@ -0,0 +1,23 @@
+/**
+ * Copyright (C) 2015 BonitaSoft S.A.
```
Comment (apupier) [inline backend/webapp/src/main/java/org/bonitasoft/web/designer/repository/exception/NotAllowedException.java]: BonitaSoft --> Bonitasoft

File: backend/webapp/src/main/java/org/bonitasoft/web/designer/repository/exception/NotAllowedException.java [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+ * BonitaSoft, 32 rue Gustave Eiffel - 38000 Grenoble
```
Comment (apupier) [inline backend/webapp/src/main/java/org/bonitasoft/web/designer/repository/exception/NotAllowedException.java]: BonitaSoft --> Bonitasoft

File: backend/webapp/src/main/java/org/bonitasoft/web/designer/repository/exception/NotFoundException.java [hunk 11]
```diff
@@ -0,0 +1,22 @@
+/**
+ * Copyright (C) 2015 BonitaSoft S.A.
```
Comment (apupier) [inline backend/webapp/src/main/java/org/bonitasoft/web/designer/repository/exception/NotFoundException.java]: BonitaSoft --> Bonitasoft

File: backend/webapp/src/main/java/org/bonitasoft/web/designer/repository/exception/NotFoundException.java [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+ * BonitaSoft, 32 rue Gustave Eiffel - 38000 Grenoble
```
Comment (apupier) [inline backend/webapp/src/main/java/org/bonitasoft/web/designer/repository/exception/NotFoundException.java]: BonitaSoft --> Bonitasoft

File: backend/contract/src/main/java/org/bonitasoft/web/designer/model/contract/AbstractContractInput.java [hunk 13]
```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright (C) 2015 BonitaSoft S.A.
+ * BonitaSoft, 32 rue Gustave Eiffel - 38000 Grenoble
+ * This library is free software; you can redistribute it and/or modify it under the terms
+ * of the GNU Lesser General Public License as published by the Free Software Foundation
+ * version 2.1 of the License.
+ * This library is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
```
Comment (apupier) [inline backend/contract/src/main/java/org/bonitasoft/web/designer/model/contract/AbstractContractInput.java]: Sure that it is LGPL? Only Engine is LGPL currently, it is GPL for other components

File: backend/webapp/src/test/java/org/bonitasoft/web/designer/utils/ObjectMapperExceptField.java [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+import com.fasterxml.jackson.databind.ser.impl.SimpleBeanPropertyFilter;
+import com.fasterxml.jackson.databind.ser.impl.SimpleFilterProvider;
+
+/**
+ * Custom jackson mapper that do not serialize property passed in argument
+ * <p>Example: if you do not want to serialize "template" property from widget object, create a new ObjectMapperExceptField(Widget.class, "template")</p>
+ *
+ * @author Colin Puy
+ */
+@SuppressWarnings("serial")
```
Comment (javamind) [inline backend/webapp/src/test/java/org/bonitasoft/web/designer/utils/ObjectMapperExceptField.java]: Used only in Eclipse

**Comment by javamind** [comment]
:+1:  the server works fine

File: backend/webapp/src/main/generator/js/generator.js (hunk 3 above)
Comment (cpuy) [inline backend/webapp/src/main/generator/js/generator.js]: Yep this is like that for now, we can change it in a further PR

File: backend/webapp/src/main/java/org/bonitasoft/web/designer/experimental/parametrizedWidget/DatePickerWidget.java (hunk 5 above)
Comment (cpuy) [inline backend/webapp/src/main/java/org/bonitasoft/web/designer/experimental/parametrizedWidget/DatePickerWidget.java]: Will be changed while contract generation is refactored

File: backend/webapp/src/main/generator/js/generator.js (hunk 3 above)
Comment (javamind) [inline backend/webapp/src/main/generator/js/generator.js]: :+1:

File: backend/contract/src/main/java/org/bonitasoft/web/designer/model/contract/AbstractContractInput.java (hunk 13 above)
Comment (cpuy) [inline backend/contract/src/main/java/org/bonitasoft/web/designer/model/contract/AbstractContractInput.java]: I'm not sure, this has been copied from engine

File: backend/webapp/src/main/java/org/bonitasoft/web/designer/experimental/parametrizedWidget/DatePickerWidget.java (hunk 5 above)
Comment (javamind) [inline backend/webapp/src/main/java/org/bonitasoft/web/designer/experimental/parametrizedWidget/DatePickerWidget.java]: :+1:

File: backend/contract/src/main/java/org/bonitasoft/web/designer/model/contract/AbstractContractInput.java (hunk 13 above)
Comment (apupier) [inline backend/contract/src/main/java/org/bonitasoft/web/designer/model/contract/AbstractContractInput.java]: ok unless there is a decision to provide it in LGPL by default it should be GPL.

**Comment by cpuy** [comment]
Ok I've changed this PR according to your comments.

I've to figure out what are the copyright that have to be included in source code before we can merge

**Comment by cpuy** [comment]
Copyright changed

File: backend/contract/src/main/java/org/bonitasoft/web/designer/model/contract/AbstractContractInput.java [hunk 15]
```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright (C) 2015 Bonitasoft S.A.
+ * Bonitasoft, 31 rue Gustave Eiffel - 38000 Grenoble
```
Comment (apupier) [inline backend/contract/src/main/java/org/bonitasoft/web/designer/model/contract/AbstractContractInput.java]: 31 --> 32 rue Gustave Eiffel

**Comment by cpuy** [comment]
retest this please

**Comment by cpuy** [comment]
retest this please

**Comment by cpuy** [comment]
retest this please

**Comment by cpuy** [comment]
king-of-the-world

**Pull request merged**